compiler-plugin/src/main/java/io/ballerina/stdlib/ftp/plugin/FtpCodeTemplateWithCaller.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.ftp.plugin;
+
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.projects.plugins.codeaction.CodeAction;
+import io.ballerina.projects.plugins.codeaction.CodeActionArgument;
+import io.ballerina.projects.plugins.codeaction.CodeActionContext;
+import io.ballerina.projects.plugins.codeaction.CodeActionExecutionContext;
+import io.ballerina.projects.plugins.codeaction.CodeActionInfo;
+import io.ballerina.projects.plugins.codeaction.DocumentEdit;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.text.LineRange;
+import io.ballerina.tools.text.TextDocumentChange;
+import io.ballerina.tools.text.TextEdit;
+import io.ballerina.tools.text.TextRange;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static io.ballerina.stdlib.ftp.plugin.PluginConstants.CODE_TEMPLATE_NAME_WITH_CALLER;
+import static io.ballerina.stdlib.ftp.plugin.PluginConstants.LS;
+import static io.ballerina.stdlib.ftp.plugin.PluginConstants.NODE_LOCATION;
+import static io.ballerina.stdlib.ftp.plugin.PluginUtils.findNode;
+
+/**
+ * Code action to add resource code snippet.
+ */
+public class FtpCodeTemplateWithCaller implements CodeAction {
+
+    private static final String REMOTE_FUNCTION_TEXT = LS + "\tremote function onFileChange(ftp:WatchEvent & " +
+            "readonly event, ftp:Caller caller) returns ftp:Error? {" + LS + LS + "\t}" + LS;
```

Comment:
Shall we switch the `ftp:Caller caller`. and `ftp:WatchEvent readonly event` parameters? Nothing wrong here. But most of the time we write the `ftp:Caller caller` as the first parameter.

Reply:
Updated in https://github.com/ballerina-platform/module-ballerina-ftp/pull/580/commits/e2109f21a68f500cd4c9ff9d3d2e08a26824133d